content/articles/graphs-in-data-structure -using _c++/index.md

```diff
@@ -0,0 +1,153 @@
+### Graphs in Data structure ( Using C++)
+
+#### Introduction
+Graphs are mathematical structures that reflect the pairwise relationship between things. A graph is a type of flow structure that displays the interactions of several objects. It may be represented by utilizing the two fundamental components shown below:
+
+* **Nodes**: These are the most crucial elements of every graph. Edges are used to represent node connections. A graph that have two nodes connected to each other using an undirected edge shows a bi-directional connection between those two nodes.
+
+* **Edges**: Edges are part of a graph showing the connections between nodes. An edge represent the connection between two nodes.
+
+#### An application in real life 
+
+* **Google Maps**: link your journey from the start to the end
+* **Social Networks**: Friends with each other are connected using an edge where each user represents a vertex.
+* **Recommendation System**: relationship data between users recommendation is use graph for connection.
+
+#### Type of Graphs :
+* **Undirected**: All edges in an undirected graph are bidirectional and do not point in any particular direction.
+
+![graph](./graph_1.png)
+
+* **Directed**: In a directed graph all of the edges are uni-directional they point to a single direction.
+
+![graph](./graph__2.JPG)
+
+* **Weighted**: Weighted graph, with each edge, allocated a weight or cost. Consider a network with four nodes, shown in the picture. Every edge of a graph have some weight with it, if you wish to get from vertex 2 to vertex 1, you can choose one of three routes:
```

Comment:
Change "Every edge of a graph have some weight with it, if you wish to get from vertex 2 to vertex 1, you can choose one of three routes:" to "Every edge of a graph has some weight with it. If you wish to get from vertex 2 to vertex 1, you can choose one of three routes:	 "